### PR TITLE
Adding zone spawn to the FNPCs

### PR DESCRIPTION
**Pull request by BaileyS03**
# Description

This pull request implements spawning entities in specific areas.  This also changed the AMOUNT of entities spawns as well since they are far more spread out.

Closes (#562)

## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?
- [x] Running previous unit tests with cover the code
- [x] Visual inspection of the entities spawning in the right area

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by mitsudz — COMMENTED** [review]
Looks good, but can you use the location/zone (I don't remember what it's called) enum already defined in the areas/terrain package or define another enum rather than using magic numbers (first options is probably better imo).

File: source/core/src/main/com/csse3200/game/areas/forest/ForestGameArea.java [hunk 1]
```diff
@@ -527,31 +527,31 @@ private void spawnFriendlyNPCs() {
...
 
         // Turtle
         generator = () -> NPCFactory.createTurtle(player, this.enemies);
-        spawnRandomNPC(generator, ForestSpawnConfig.NUM_TURTLES);
+        spawnRandomNPC(generator, ForestSpawnConfig.NUM_TURTLES, 2);
 
         // Eagle
         generator = () -> NPCFactory.createEagle(player, this.enemies);
-        spawnRandomNPC(generator, ForestSpawnConfig.NUM_EAGLES);
+        spawnRandomNPC(generator, ForestSpawnConfig.NUM_EAGLES, 2);
```
Comment (LucasRyall) [inline source/core/src/main/com/csse3200/game/areas/forest/ForestGameArea.java:546]: Shouldn't this (and the magpie) be in zone 3?

**Review by LucasRyall — APPROVED** [review]
Simple change, looks good.

**Comment by BaileyS03** [comment]
> Looks good, but can you use the location/zone (I don't remember what it's called) enum already defined in the areas/terrain package or define another enum rather than using magic numbers (first options is probably better imo).

Yes ! I didn't know this existed and will implement this 
Do you mean this enum ?

```java
  /**
   * Map types
   */
  public enum MapType {
    FOREST, WATER, AIR, COMBAT, FOG, MAZE_MINIGAME, NONE
  }
  ```

File: source/core/src/main/com/csse3200/game/areas/forest/ForestGameArea.java (hunk 1 above)
Comment (BaileyS03) [reply source/core/src/main/com/csse3200/game/areas/forest/ForestGameArea.java]: Thank you! I saw the bees were in zone 2 and assumed that was air, but this is now fixed.

**Comment by LucasRyall** [comment]
> > Looks good, but can you use the location/zone (I don't remember what it's called) enum already defined in the areas/terrain package or define another enum rather than using magic numbers (first options is probably better imo).
> 
> Yes ! I didn't know this existed and will implement this Do you mean this enum ?
> 
> ```java
>   /**
>    * Map types
>    */
>   public enum MapType {
>     FOREST, WATER, AIR, COMBAT, FOG, MAZE_MINIGAME, NONE
>   }
> ```

I think he might have been talking about the enum defined in this file (i could be wrong cause there are 2 others in this folder) source/core/src/main/com/csse3200/game/areas/terrain/enums/TileLocation.java

**Comment by mitsudz** [comment]
> > Looks good, but can you use the location/zone (I don't remember what it's called) enum already defined in the areas/terrain package or define another enum rather than using magic numbers (first options is probably better imo).
> 
> Yes ! I didn't know this existed and will implement this Do you mean this enum ?
> 
> ```java
>   /**
>    * Map types
>    */
>   public enum MapType {
>     FOREST, WATER, AIR, COMBAT, FOG, MAZE_MINIGAME, NONE
>   }
> ```

Yes that one seems reasonable to use. If you look in the areas/terrain/enums package I had (with approval from @Howard-12 from team 8 who worked on this) moved a lot of the enums there - might be good to refactor this to move this enum out of the map handler file to that package so it can be reused more easily (there might also already be an enum there with this sort of functionality enabled - so maybe look in there to see if there's already a better one).

**Comment by mitsudz** [comment]
> > > Looks good, but can you use the location/zone (I don't remember what it's called) enum already defined in the areas/terrain package or define another enum rather than using magic numbers (first options is probably better imo).
> > 
> > 
> > Yes ! I didn't know this existed and will implement this Do you mean this enum ?
> > ```java
> >   /**
> >    * Map types
> >    */
> >   public enum MapType {
> >     FOREST, WATER, AIR, COMBAT, FOG, MAZE_MINIGAME, NONE
> >   }
> > ```
> 
> I think he might have been talking about the enum defined in this file (i could be wrong cause there are 2 others in this folder) source/core/src/main/com/csse3200/game/areas/terrain/enums/TileLocation.java

Yeah I think the two referenced in this thread can probably be merged together - I had meant the one you said @LucasRyall - but it seems redundant to have both.

**Review by Howard-12 — APPROVED** [review]

**Review by mitsudz — APPROVED** [review]
Looks great!

**Pull request merged**